README.md
一些或实用或有趣的小练习、小项目，帮你提升编程技能

欢迎关注“**Crossin的编程教室**”及同名 [知乎专栏](https://zhuanlan.zhihu.com/crossin)

![crossincode](crossin-logo.png)



sports/history.txt




1  ┃ 山东  ┃ 65  ┃ 47.5 ┃ 43  ┃ 155.5┃ 11.5 ┃ 6  ┃ 2
2  ┃ 辽宁  ┃ 56  ┃ 49.5 ┃ 39.5 ┃ 145  ┃ 4   ┃ 2  ┃ 2.5
3  ┃ 广东  ┃ 50.5 ┃ 34.5 ┃ 36  ┃ 121  ┃ 9.5 ┃ 6  ┃ 4
4  ┃ 上海  ┃ 45  ┃ 48 ┃ 36.5 ┃ 129.5 ┃ 6 ┃ 9 ┃ 4.5
5   ┃ 解放军 ┃ 45 ┃ 38 ┃ 37 ┃ 120 ┃ 10 ┃ 4 ┃ 0
5   ┃ 江苏  ┃ 45 ┃ 35 ┃ 37 ┃ 117 ┃ 7 ┃ 3 ┃ 4
6   ┃ 浙江  ┃ 35 ┃ 20 ┃ 39 ┃ 94 ┃ 10 ┃ 3 ┃ 3
7  ┃ 北京  ┃ 33 ┃ 28 ┃ 25 ┃ 86 ┃ 4 ┃ 7 ┃ 1
8  ┃ 黑龙江 ┃ 22 ┃ 15 ┃ 29 ┃ 66 ┃ 11 ┃ 3 ┃ 11
9  ┃ 天津  ┃ 20 ┃ 14 ┃ 11 ┃ 45 ┃ 5 ┃ 2 ┃ 2
10 ┃ 福建  ┃ 18 ┃ 23 ┃ 22.5 ┃ 63.5 ┃ 4 ┃ 4 ┃ 6
11 ┃ 湖南  ┃ 15 ┃ 13 ┃ 13 ┃ 41 ┃ 2 ┃ 1 ┃ 2
12 ┃ 河南  ┃ 14.5 ┃ 7 ┃ 18.5 ┃ 40 ┃ 1 ┃ 0 ┃ 0.5
13 ┃ 四川  ┃ 13 ┃ 26 ┃ 27 ┃ 66 ┃ 6 ┃ 5 ┃ 4
14 ┃ 吉林  ┃ 13 ┃ 17 ┃ 17 ┃ 47 ┃ 5 ┃ 7 ┃ 3
15 ┃ 山西  ┃ 10 ┃ 8 ┃ 6 ┃ 24 ┃ 1 ┃ 1 ┃ 2
16 ┃ 陕西  ┃ 9 ┃ 11 ┃ 11 ┃ 31 ┃ 3 ┃ 3 ┃ 0
17 ┃ 安徽  ┃ 8.5 ┃ 9 ┃ 12 ┃ 29.5 ┃ 2 ┃ 0 ┃ 0
18 ┃ 湖北  ┃ 8 ┃ 18.5 ┃ 14 ┃ 40.5 ┃ 2 ┃ 0 ┃ 0
19 ┃ 河北  ┃ 6 ┃ 10 ┃ 28 ┃ 44 ┃ 0 ┃ 1 ┃ 3
20 ┃ 新疆  ┃ 6 ┃ 8 ┃ 2 ┃ 16 ┃ 1 ┃ 1 ┃ 0
21 ┃ 江西  ┃ 6 ┃ 5 ┃ 5.5 ┃ 16.5 ┃ 1 ┃ 0 ┃ 0
22 ┃ 内蒙古 ┃ 5 ┃ 6 ┃ 11 ┃ 22 ┃ 0 ┃ 0 ┃ 0
23  ┃ 重庆  ┃ 5 ┃ 3 ┃ 4 ┃ 12 ┃ 2 ┃ 0 ┃ 0
24 ┃ 云南  ┃ 4 ┃ 8 ┃ 4 ┃ 16 ┃ 1 ┃ 0 ┃ 1
25 ┃ 广西  ┃ 4 ┃ 5 ┃ 5 ┃ 14 ┃ 0 ┃ 0 ┃ 0
26 ┃ 青海  ┃ 3 ┃ 1 ┃ 2 ┃ 6 ┃ 0 ┃ 0 ┃ 2
27 ┃ 海南  ┃ 3 ┃ 1 ┃ 1 ┃ 5 ┃ 0 ┃ 0 ┃ 0
28 ┃ 甘肃  ┃ 2 ┃ 4 ┃ 1 ┃ 7 ┃ 0 ┃ 0 ┃ 0
29 ┃ 前卫  ┃ 2 ┃ 3 ┃ 3 ┃ 8 ┃ 0 ┃ 1 ┃ 0
30 ┃ 贵州  ┃ 1 ┃ 4 ┃ 6 ┃ 11 ┃ 1 ┃ 0 ┃ 0
31 ┃ 香港  ┃ 1 ┃ 4 ┃ 3 ┃ 8 ┃ 0 ┃ 0 ┃ 0
32 ┃ 西藏  ┃ 1 ┃ 1 ┃ 0 ┃ 2 ┃ 0 ┃ 0 ┃ 0
33 ┃ 宁夏  ┃ 0 ┃ 0 ┃ 2 ┃ 2 ┃ 0 ┃ 0 ┃ 0
34 ┃ 新疆兵 ┃ 0 ┃ 0 ┃ 1 ┃ 1 ┃ 0 ┃ 0 ┃ 0
34 ┃ 火车头 ┃ 0 ┃ 0 ┃ 1 ┃ 1 ┃ 0 ┃ 0 ┃ 0
35 ┃ 澳门  ┃ 0 ┃ 0 ┃ 0 ┃ 0 ┃ 0 ┃ 0 ┃ 0
35  ┃ 煤矿  ┃ 0 ┃ 0 ┃ 0 ┃ 0 ┃ 0 ┃ 0 ┃ 0

 

 





 1 ┃ 山东 ┃ 63  ┃ 44  ┃ 46  ┃ 10 ┃ 4 ┃ 3
 2 ┃解放军┃ 49  ┃ 39  ┃ 41  ┃ 13 ┃ 9 ┃ 8
 3 ┃ 江苏 ┃ 48.5 ┃ 37  ┃ 36  ┃ 16 ┃ 6 ┃ 6
 4 ┃ 辽宁 ┃ 48  ┃ 45  ┃ 34  ┃ 13 ┃ 12┃ 8
 5 ┃ 广东 ┃ 45  ┃ 45.5 ┃ 39.5 ┃ 11 ┃ 8 ┃ 6
 6 ┃ 上海 ┃ 41  ┃ 34  ┃ 46.5 ┃ 10 ┃ 3 ┃15
 7 ┃ 北京 ┃ 30  ┃ 20  ┃ 29  ┃ 9  ┃ 4 ┃ 8
 8 ┃黑龙江┃ 23.5 ┃ 23  ┃ 18.5 ┃ 6 ┃ 9 ┃ 8
 9 ┃ 天津 ┃ 23  ┃ 14  ┃ 15.5 ┃ 6  ┃ 2 ┃ 5
10 ┃ 福建 ┃ 19  ┃ 11  ┃ 21  ┃ 6 ┃ 0 ┃ 2
11 ┃ 湖南 ┃ 18  ┃ 10.5 ┃ 6   ┃ 5  ┃ 0 ┃ 0
12 ┃ 浙江 ┃ 16  ┃ 27  ┃ 25  ┃ 3 ┃ 5 ┃ 1
13 ┃ 四川 ┃ 13.5 ┃ 21  ┃ 26  ┃ 7  ┃ 3 ┃ 4
14 ┃ 河北 ┃ 13  ┃ 13  ┃ 19.5 ┃ 5 ┃ 0 ┃ 0
15 ┃ 安徽 ┃ 13  ┃ 12.5 ┃ 9   ┃ 1  ┃ 4 ┃ 0
16 ┃ 湖北 ┃ 12  ┃ 7.5  ┃ 15  ┃ 6 ┃ 2 ┃ 6
17 ┃ 江西 ┃ 10.5 ┃ 7.5  ┃ 7   ┃ 3  ┃ 2 ┃ 1
18 ┃ 山西 ┃ 9   ┃  7  ┃ 12  ┃ 0 ┃ 2 ┃ 2
19 ┃ 陕西 ┃ 9   ┃  6  ┃ 4.5  ┃ 3  ┃ 0 ┃ 2
20 ┃ 吉林 ┃ 7.5  ┃ 10  ┃ 12  ┃ 1  ┃ 3 ┃ 4
21 ┃ 河南 ┃ 7.5 ┃ 6.5  ┃ 8   ┃ 0  ┃ 2 ┃ 0
22 ┃ 广西 ┃ 7.5 ┃ 4   ┃ 3.5  ┃ 3  ┃ 0 ┃ 1
23 ┃内蒙古┃ 7   ┃ 4.5  ┃ 8.5  ┃ 2 ┃ 0 ┃ 0
24 ┃ 贵州 ┃ 3   ┃ 5   ┃ 3   ┃ 1 ┃ 0 ┃ 0
25 ┃ 云南 ┃ 3   ┃ 4.5  ┃ 4   ┃ 0  ┃ 0 ┃ 0
26 ┃ 前卫 ┃ 2   ┃ 4   ┃ 2   ┃ 0 ┃ 0 ┃ 0
27 ┃火车头┃ 2   ┃ 2   ┃ 2    ┃ 0 ┃ 0 ┃ 0
28 ┃ 香港 ┃ 2   ┃ 1   ┃ 4   ┃ 0  ┃ 0 ┃ 0
29 ┃ 重庆 ┃ 2   ┃ 0   ┃ 5   ┃ 0  ┃ 0 ┃ 1
30 ┃ 新疆 ┃ 1.5 ┃ 5.5  ┃ 7.5  ┃ 0 ┃ 1 ┃ 3
31 ┃ 甘肃 ┃ 1   ┃ 2.5  ┃ 7   ┃ 0 ┃ 0 ┃ 0
32 ┃ 宁夏 ┃ 1   ┃ 1   ┃ 4   ┃ 0  ┃ 0 ┃ 0
33 ┃ 青海 ┃ 0   ┃ 1   ┃ 2   ┃ 0  ┃ 0 ┃ 0
34 ┃ 西藏 ┃ 0   ┃ 0   ┃ 2.5  ┃ 0  ┃ 0 ┃ 0
35 ┃ 海南 ┃ 0   ┃ 0   ┃ 2   ┃ 0 ┃ 0 ┃ 0
36 ┃ 通信 ┃ 0   ┃ 0   ┃ 2   ┃ 0  ┃ 0 ┃ 0
 


1 ┃ 江苏 ┃ 56 ┃ 38 ┃ 42 ┃ 5 ┃ 1 ┃ 2
2 ┃ 广东 ┃ 46 ┃ 42.5┃ 36.5 ┃ 9 ┃ 4 ┃ 2
3 ┃解放军┃ 44 ┃ 39 ┃ 32 ┃ 4 ┃ 4 ┃ 2
4 ┃ 山东 ┃ 42 ┃ 29 ┃ 27 ┃ 10 ┃ 5 ┃ 0
5 ┃ 北京 ┃ 32 ┃ 23.5 ┃ 26 ┃ 10 ┃ 2 ┃ 2
6 ┃ 辽宁 ┃ 31 ┃ 35 ┃ 33 ┃ 7 ┃ 2 ┃ 7
7 ┃ 浙江 ┃ 29 ┃ 20 ┃ 12 ┃ 7 ┃ 1 ┃ 2
8 ┃ 上海 ┃ 26 ┃ 48 ┃ 44.5 ┃ 3 ┃ 7 ┃ 2
9 ┃ 福建 ┃ 17 ┃ 12 ┃ 14 ┃ 2 ┃ 3 ┃ 2
10 ┃黑龙江┃ 16 ┃ 17 ┃ 8.5 ┃ 4 ┃ 1 ┃ 2
11 ┃ 天津 ┃ 15.5 ┃ 13.5 ┃ 11 ┃ 2 ┃ 2 ┃ 0
12 ┃ 河南 ┃ 15.5 ┃ 7 ┃ 18.5 ┃ 5 ┃ 0 ┃ 0
13 ┃ 河北 ┃ 15 ┃ 9 ┃ 11 ┃ 3 ┃ 0 ┃ 1
14 ┃ 湖南 ┃ 13 ┃ 12 ┃ 8 ┃ 0 ┃ 3 ┃ 2
15 ┃ 四川 ┃ 12 ┃ 17 ┃ 19 ┃ 1 ┃ 2 ┃ 1
16 ┃ 江西 ┃ 12 ┃ 6 ┃ 2 ┃ 3 ┃ 0 ┃ 0
17 ┃ 吉林 ┃ 10 ┃ 17 ┃ 11.5 ┃ 0 ┃ 7 ┃ 6
18 ┃ 山西 ┃ 10 ┃ 5 ┃ 5 ┃ 0 ┃ 0 ┃ 0
19 ┃ 安徽 ┃ 7 ┃ 5 ┃ 7 ┃ 0 ┃ 0 ┃ 0
20 ┃ 湖北 ┃ 7 ┃ 4.5 ┃ 8 ┃ 2 ┃ 1 ┃ 0
21 ┃ 广西 ┃ 6 ┃ 8 ┃ 8 ┃ 1 ┃ 0 ┃ 2
22 ┃ 云南 ┃ 5.5 ┃ 8 ┃ 8.5 ┃ 2 ┃ 1 ┃ 0
23 ┃ 陕西 ┃ 3 ┃ 7.5 ┃ 8.5 ┃ 1 ┃ 2 ┃ 2
24 ┃ 甘肃 ┃ 2 ┃ 4 ┃ 1 ┃ 0 ┃ 0 ┃ 0
25 ┃ 火车头 ┃ 2 ┃ 1 ┃ 1 ┃ 0 ┃ 0 ┃ 0
26 ┃ 新疆 ┃ 1.5 ┃ 2.5 ┃ 6.5 ┃ 0 ┃ 0 ┃ 0
27 ┃ 内蒙古 ┃ 1 ┃ 4.5 ┃ 10 ┃ 0 ┃ 0 ┃ 0
28 ┃ 重庆 ┃ 1 ┃ 3 ┃ 0.5 ┃ 0 ┃ 0 ┃ 0
29 ┃ 个人 ┃ 1 ┃ 2 ┃ 2 ┃ 0 ┃ 0 ┃ 0
30 ┃ 海南 ┃ 1 ┃ 1 ┃ 4 ┃ 0 ┃ 0 ┃ 0
31 ┃ 贵州 ┃ 1 ┃ 0.5 ┃ 3.5 ┃ 0 ┃ 0 ┃ 2
32 ┃ 香港 ┃ 1 ┃ 0 ┃ 3 ┃ 0 ┃ 0 ┃ 0
33 ┃ 前卫 ┃ 1 ┃ 0 ┃ 2 ┃ 0 ┃ 0 ┃ 0
34 ┃ 宁夏 ┃ 0 ┃ 1.5 ┃ 0.5 ┃ 0 ┃ 0 ┃ 0
35 ┃ 新疆兵团 ┃ 0 ┃ 1 ┃ 0.5 ┃ 0 ┃ 0 ┃ 0
36 ┃ 煤矿 ┃ 0 ┃ 1 ┃ 0 ┃ 0 ┃ 0 ┃ 0
37 ┃ 青海 ┃ 0 ┃ 0 ┃ 2 ┃ 0 ┃ 0 ┃ 0
38 ┃ 通信 ┃ 0 ┃ 0 ┃ 1 ┃ 0 ┃ 0 ┃ 0

 


1 ┃ 广东 ┃ 69.5 ┃ 48 ┃ 51.5      
2 ┃ 辽宁 ┃ 41 ┃ 31.5 ┃ 28.5      
3 ┃ 解放军 ┃ 33 ┃ 26 ┃ 28      
4 ┃ 上海 ┃ 29.5 ┃ 29.5 ┃ 24.5      
5 ┃ 江苏 ┃ 24.5 ┃ 21.5 ┃ 25      
6 ┃ 北京 ┃ 23 ┃ 27 ┃ 22.5      
7 ┃ 山东 ┃ 22 ┃ 22 ┃ 22      
8 ┃ 湖南 ┃ 19.5 ┃ 14.5 ┃ 17      
9 ┃ 浙江 ┃ 12.5 ┃ 12.5 ┃ 17.5      
10 ┃ 湖北 ┃ 12.5 ┃ 6.5 ┃ 11      
11 ┃ 吉林 ┃ 12 ┃ 17 ┃ 10      
12 ┃ 河北 ┃ 12 ┃ 10.5 ┃ 10.5      
13 ┃ 四川 ┃ 9 ┃ 16 ┃ 21.5      
14 ┃ 福建 ┃ 9 ┃ 9 ┃ 12.5      
15 ┃ 黑龙江 ┃ 8.5 ┃ 10 ┃ 12      
16 ┃ 山西 ┃ 8.5 ┃ 7 ┃ 2      
17 ┃ 陕西 ┃ 8.5 ┃ 4.5 ┃ 4      
18 ┃ 天津 ┃ 8 ┃ 8 ┃ 9      
19 ┃ 河南 ┃ 7.5 ┃ 18.5 ┃ 10      
20 ┃ 广西 ┃ 7 ┃ 7.5 ┃ 7.5      
21 ┃ 云南 ┃ 7 ┃ 7.5 ┃ 5.5      
22 ┃ 内蒙古 ┃ 6.5 ┃ 2.5 ┃ 5.5      
23 ┃ 安徽 ┃ 3.5 ┃ 8.5 ┃ 11      
24 ┃ 甘肃 ┃ 3.5 ┃ 4.5 ┃ 6.5      
25 ┃ 江西 ┃ 3 ┃ 9.5 ┃ 5.5      
26 ┃ 新疆 ┃ 2 ┃ 5 ┃ 2      
27 ┃ 香港 ┃ 2 ┃ 2 ┃ 1      
28 ┃ 前卫 ┃ 2 ┃ 0 ┃ 2      
29 ┃ 贵州 ┃ 1.5 ┃ 2.5 ┃ 4.5      
30 ┃ 重庆 ┃ 1.5 ┃ 2.5 ┃ 3      
31 ┃ 海南 ┃ 1 ┃ 2.5 ┃ 2      
32 ┃ 火车头 ┃ 1 ┃ 1.5 ┃ 1.5      
33 ┃ 青海 ┃ 0 ┃ 1 ┃ 2      
34 ┃ 西藏 ┃ 0 ┃ 1 ┃ 0.5      
35 ┃ 宁夏 ┃ 0 ┃ 0.5 ┃ 3      
36 ┃ 新疆生产建设兵团 ┃ 0 ┃ 0 ┃ 2      
37 ┃ 金融 ┃ 0 ┃ 0 ┃ 1      


 


1 ┃ 上海 ┃ 42 ┃ 34 ┃ 32      
2 ┃ 辽宁 ┃ 39.5 ┃ 43.5 ┃ 29.5      
3 ┃ 山东 ┃ 25 ┃ 22 ┃ 25      
4 ┃ 广东 ┃ 24.5 ┃ 31.5 ┃ 37.5      
5 ┃ 解放军 ┃ 24.5 ┃ 25.5 ┃ 24.5      
6 ┃ 江苏 ┃ 23 ┃ 15 ┃ 24      
7 ┃ 北京 ┃ 20 ┃ 33 ┃ 12.5      
8 ┃ 吉林 ┃ 19 ┃ 9.5 ┃ 6      
9 ┃ 湖南 ┃ 17.5 ┃ 10 ┃ 13.5      
10 ┃ 浙江 ┃ 17 ┃ 10 ┃ 7.5      
11 ┃ 河南 ┃ 14.5 ┃ 11 ┃ 20      
12 ┃ 四川 ┃ 12.5 ┃ 20.5 ┃ 18      
13 ┃ 广西 ┃ 12.5 ┃ 11 ┃ 7.5      
14 ┃ 湖北 ┃ 12.5 ┃ 10.5 ┃ 9      
15 ┃ 黑龙江 ┃ 9.5 ┃ 11 ┃ 8      
16 ┃ 内蒙古 ┃ 9.5 ┃ 9 ┃ 9      
17 ┃ 山西 ┃ 8 ┃ 7 ┃ 8      
18 ┃ 江西 ┃ 8 ┃ 4 ┃ 3      
19 ┃ 河北 ┃ 6.5 ┃ 5.5 ┃ 17.5      

exam/exam.py

# coding: utf-8

# In[50]:


data = [[2017, 940, 700, 74.46],
[2016, 940, 705, 75],
[2015, 942, 700, 74.3],
[2014, 939, 698, 74.3],
[2013, 912, 694, 76],
[2012, 915, 685, 75],
[2011, 933, 675, 72],
[2010, 946, 657, 69],
[2009, 1020, 629, 62],
[2008, 1050, 599, 57],
[2007, 1010, 566, 56],
[2006, 950, 546, 57],
[2005, 877, 504, 57],
[2004, 729, 447, 61],
[2003, 613, 382, 62],
[2002, 510, 320, 63],
[2001, 454, 268, 59],
[2000, 375, 221, 59],
[1999, 288, 160, 56],
[1998, 320, 108, 34],
[1997, 278, 100, 36],
[1996, 241, 97, 40],
[1995, 253, 93, 37],
[1994, 251, 90, 36],
[1993, 286, 98, 34],
[1992, 303, 75, 25],
[1991, 296, 62, 21],
[1990, 283, 61, 22],
[1989, 266, 60, 23],
[1988, 272, 67, 25],
[1987, 228, 62, 27],
[1986, 191, 57, 30],
[1985, 176, 62, 35],
[1984, 164, 48, 29],
[1983, 167, 39, 23],
[1982, 187, 32, 17],
[1981, 259, 28, 11],
[1980, 333, 28, 8],
[1979, 468, 28, 6],
[1978, 610, 40.2, 7],
[1977, 570, 27, 5]]


# In[51]:


import numpy as np
arr = np.array(data)


# In[57]:


from pyecharts import Bar, Line, Overlap

bar = Bar("历年报考人数与录取率")
bar.add("报考人数",arr[::-1,0] , arr[::-1,1])
bar.add("录取人数",arr[::-1,0] , arr[::-1,2])
line = Line()
line.add("录取率",arr[::-1,0] , arr[::-1,3]*10, line_width=3, line_color='green')
ol = Overlap()
ol.add(bar)
ol.add(line)
ol


# In[112]:


data_born = [[124761, 123626, 122389, 121121, 119850, 118517, 117171, 115823, 114333, 112704, 111026, 109300, 107507, 105851, 104357, 103008, 101654, 100072, 98705, 97542, 96259, 94974, 93717, 92420, 90859, 89211, 87177, 85229, 82992, 80671, 78534, 76368, 74542, 72538, 70499, 69172, 67296, 65859, 66207, 67207, 65994], 
             [14.64, 15.64, 16.57, 16.98, 17.12, 17.70, 18.09, 18.24, 19.68, 21.06, 21.58, 22.37, 23.33, 22.43, 21.04, 19.90, 20.19, 22.28, 20.91, 18.21, 17.82, 18.25, 19.03, 20.01, 23.13, 24.95, 28.07, 29.92, 30.74, 33.59, 34.25, 35.75, 34.12, 35.21, 38.00, 39.34, 43.60, 37.22, 18.13, 20.86, 24.78]]
arr_born = np.array(data_born)
list_born = list((arr_born[0]*arr_born[1]/1000)[::-1])
list_born


# In[113]:


from pyecharts import Bar, Line, Overlap

bar = Bar("历年报考人数与出生人数")
bar.add("报考人数",arr[::-1,0] , arr[::-1,1])
bar.add("出生人数",arr[::-1,0] , list_born)
bar


# In[59]:


from pyecharts import Map  
  
data_prov = [('北京', 6.3), ('重庆', 25), ('上海', 5), ('天津', 5.5), ('河南', 98.38), ('广东', 75.8), ('山东', 59.2), ('安徽', 49.9), ('四川', 62), ('江苏', 33.15), ('河北', 48.64), ('湖北', 37.4), ('陕西', 31.9), ('湖南', 45.18), ('山西', 30.5), ('浙江', 30.6), ('甘肃', 27.3), ('广西', 40), ('江西', 38), ('福建', 20.09), ('辽宁', 18.5), ('贵州', 44), ('云南', 30), ('黑龙江', 19.04), ('内蒙古', 19.5), ('吉林', 15.03), ('新疆', 20.74), ('宁夏', 6.95), ('海南', 5.88), ('青海', 4), ('西藏', 2.53)]  
geo = Map("2018各省高考报名人数")
attr, value = geo.cast(data_prov)
geo.add("", attr, value, visual_range=[0, 80], is_visualmap=True, is_map_symbol_show=False)  
geo


# In[48]:


data_prov_ad = [('湖南', 11.2), ('江西', 10.4), ('安徽', 14.2), ('广东', 11.2), ('河北', 14.55), ('湖北', 15.8), ('河南', 7.8), ('山西', 9.8), ('福建', 18.7), ('吉林', 12.4), ('重庆', 11.5), ('辽宁', 16.1), ('内蒙古', 14.03), ('甘肃', 15.22), ('黑龙江', 13.5), ('陕西', 14.6), ('宁夏', 19), ('新疆', 12.86), ('西藏', 11), ('青海', 15.2), ('四川', 10.6), ('云南', 10.85), ('广西', 8.44), ('贵州', 14.50), ('北京', 30.5), ('江苏', 12.10), ('山东', 10.6), ('天津', 24.1), ('上海', 21.8), ('浙江', 14)]
geo = Map("2018各省一本录取率")
attr, value = geo.cast(data_prov_ad)
geo.add("", attr, value, visual_range=[8, 20], is_visualmap=True, is_map_symbol_show=False)  
geo


# In[ ]:




